### PR TITLE
Staged manager uses plain Operations instead of ProvisioningOperations

### PR DESCRIPTION
**Pull request by jaroslaw-pieszka**
**Description**

Changes proposed in this pull request:

- added package with staged manager operating just on Operations - not on ProvisioningOperations
- necessary events added 
- tests adjusted

**Related issue(s)**

### CODE REVIEW

File: components/kyma-environment-broker/internal/process/new_provisioning/staged_manager.go [hunk 1]
```diff
@@ -0,0 +1,241 @@
+package new_provisioning
```
Comment (piotrmiskiewicz) [inline components/kyma-environment-broker/internal/process/new_provisioning/staged_manager.go:1]: I'd change the name of the packag. it is not about provisioning, maybe `process`? or maybe you have a better idea

File: components/kyma-environment-broker/internal/process/new_provisioning/staged_manager.go [hunk 2]
```diff
@@ -0,0 +1,241 @@
...
+		if err != nil {
+			return time.Second, nil
+		}
+	}
+
+	logOperation.Infof("Operation succeeded")
+
+	processedOperation.State = domain.Succeeded
+	m.publisher.Publish(context.TODO(), process.ProvisioningSucceeded{
+		Operation: internal.ProvisioningOperation{Operation: processedOperation},
```
Comment (piotrmiskiewicz) [inline components/kyma-environment-broker/internal/process/new_provisioning/staged_manager.go:168]: we need to send general event like OperationSucceeded, then, in handlers, we need to recognize it by field `type`

Comment (jaroslaw-pieszka) [reply components/kyma-environment-broker/internal/process/new_provisioning/staged_manager.go]: Pushed.

File: components/kyma-environment-broker/internal/metrics/operation_duration.go [hunk 3]
```diff
@@ -77,3 +77,21 @@ func (c *OperationDurationCollector) OnDeprovisioningStepProcessed(ctx context.C
...
 }
+
+func (c *OperationDurationCollector) OnOperationSucceeded(ctx context.Context, ev interface{}) error {
+	operationSucceeded, ok := ev.(process.OperationSucceeded)
+	if !ok {
+		return fmt.Errorf("expected OperationSucceeded but got %+v", ev)
+	}
+
+	if operationSucceeded.Operation.Type == "provisioning" {
+		err := c.OnProvisioningSucceeded(ctx, ev)
```
[the remark this answers is not in the document]
Comment (jaroslaw-pieszka) [reply components/kyma-environment-broker/internal/metrics/operation_duration.go]: Changes pushed

**Review by piotrmiskiewicz — APPROVED** [review]

**Pull request merged**